pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV2V3.java

```diff
@@ -167,6 +178,18 @@ public DataTableImplV2(ByteBuffer byteBuffer)
       _variableSizeDataBytes = null;
       _variableSizeData = null;
     }
+
+    // Read positional data.
+    String[] positionalData = null;
+    if (version == VERSION_3 && byteBuffer.hasRemaining()) {
+      int positionalDataStart = variableSizeDataStart + variableSizeDataLength;
+      int positionalDataLength = byteBuffer.remaining();
+      byteBuffer.position(positionalDataStart);
```

Comment:
Since we are using `byteBuffer.remaining() `to compute the length of positional data, it implies we are treating it as a **footer** of specific format (name-value pairs as defined in the enum) even though we are not calling it. So, technically no other structure can come after this as we will fail to distinguish between the length of positional data + whatever comes after it. I don't think we should limit that flexibility. Even if we call this footer, let's please write the length of footer as well before line 348

Reply:
done. Length of footer are written into header now